Add support for up script and extra config options

haslersn opened the pull request.

haslersn committed:
Add support for up script and extra config options

haslersn committed:
openvpn_client_base_dir is not supposed to be configured

The openvpn systemd service template doesn't work with base directories
other than /etc/openvpn

haslersn committed:
Adapt README to changes since a4ba024031fa

haslersn committed:
Rename ..._config_filename to ...local_config_path

The old name could suggest that it's a path on the host, while
it's actually a local path.

SlothOfAnarchy commented on README.md:
```diff
@@ -9,11 +9,25 @@ An openvpn config file either on the openvpn server or in the inventory and an a
+| Variable                           | Default                    | Description                                                                             |
+| ---------------------------------- | -------------------------- | --------------------------------------------------------------------------------------- |
+| `openvpn_client_name`              | `{{ inventory_hostname }}` | Name of the client; doesn't need to match the host name                                 |
+| `openvpn_client_server`            | `default-server`           | Ansible host name of the OpenVPN server                                                 |
```
Is `default-server` such a good default value? I find it reasonable to set a default when the role will work out-of-the-box with it, but here the user has to have a Ansible-managed server with this hostname.

haslersn replied:
No, the server name (`openvpn_client_server`) does only need to match the Ansible-managed server hostname if `openvpn_client_local_config_path` is omitted. I added an according change to the readme.

SlothOfAnarchy commented on README.md:
```diff
@@ -9,11 +9,25 @@ An openvpn config file either on the openvpn server or in the inventory and an a
+
+* Specifying a configuration file in `openvpn_client_local_config_path`.
+* Pulling the configuration file from the OpenVPN server.
+  This is done when `openvpn_client_confi_filename` is **not** specified.
```
Typo in `confi` -> `config`
Also, I think you meant `openvpn_client_local_config_path` because the variable `openvpn_client_config_filename` is not used otherwise.
```suggestion
  This is done when `openvpn_client_local_config_path` is **not** specified.
```

SlothOfAnarchy commented on README.md:
```diff
@@ -9,11 +9,25 @@ An openvpn config file either on the openvpn server or in the inventory and an a
+* Pulling the configuration file from the OpenVPN server.
+  This is done when `openvpn_client_confi_filename` is **not** specified.
+  The OpenVPN server is then required to be an Ansible host with hostname `openvpn_client_server`.
+  It needs to have the configuration file, which is to be pulled, at `/etc/openvpn/{{ openvpn_client_name }}-{{ openvpn_client_server }}.ovpn`.
```
This assumes `openvpn_base_dir` on the OpenVPN server is set to `/etc/openvpn`, just a nitpick though, I wouldn't change it. Also, the file has to be base64 encoded.

haslersn replied:
Our openvpn (server) role doesn't work if `openvpn_base_dir` is changed. This is because the official OpenVPN systemd service always assumes that directory to be `/etc/openvpn`.

SlothOfAnarchy commented on README.md:
```diff
@@ -9,11 +9,25 @@ An openvpn config file either on the openvpn server or in the inventory and an a
+
+There are two ways to configure the OpenVPN client:
+
+* Specifying a configuration file in `openvpn_client_local_config_path`.
```
Please mention that a relative path in `files/{{ inventory_hostname }}/` is expected here.

haslersn replied:
Done.

SlothOfAnarchy commented on tasks/main.yml:
```diff
@@ -1,26 +1,58 @@
+    line: script-security 2
+  notify: Restart OpenVPN service
+
+- name: Add up-script to config
```
You might be able to reduce the number of tasks by appending the two lines to the `openvpn_client_extra_config` list.

haslersn replied:
Then this would also need to be done when overwriting `openvpn_client_extra_config`.

SlothOfAnarchy replied:
Yes, I would just replace this line https://github.com/stuvusIT/openvpn-client/pull/2/files/cba0a6efd961249fe4df41dc146cc2be789d911a#diff-2444ad0870f91f17ca6c2a5e96b26823R57
with
```
with_items: "{{ openvpn_client_extra_config | combine(['script-security 2', 'up {{ openvpn_client_base_dir }}/up.sh']) }}"
```
One could also use `set_fact` or the general `vars` to create a combined version of the options.

haslersn replied:
Note: `combine` is for dictionaries.

I added an according commit.

haslersn committed:
Explain when openvpn_client_server needs to be an Ansible hostname

haslersn committed:
Fix typo that prevented a previous auto-replace

haslersn committed:
Mention that openvpn_client_local_config_path is relative

... to `files/{{ inventory_hostname }}`. This change is in reactive to a review comment:

Please mention that a relative path in `files/{{ inventory_hostname }}/` is expected here.

_Originally posted by @SlothOfAnarchy in https://github.com/stuvusIT/openvpn-client/pull/2_

haslersn committed:
More compact task code

Reactive to a review comment:

Yes, I would just replace this line https://github.com/stuvusIT/openvpn-client/pull/2/files/cba0a6efd961249fe4df41dc146cc2be789d911a#diff-2444ad0870f91f17ca6c2a5e96b26823R57
with
```
with_items: "{{ openvpn_client_extra_config | combine(['script-security 2', 'up {{ openvpn_client_base_dir }}/up.sh']) }}"
```
One could also use `set_fact` or the general `vars` to create a combined version of the options.

_Originally posted by @SlothOfAnarchy in https://github.com/stuvusIT/openvpn-client/pull/2_
